### PR TITLE
Remove memory allocator

### PR DESCRIPTION
**Pull request by MBkkt**
Firstly we never use it with non zero pool size.
So it's useless now.

Secondly it looks more like possible pessimization because internal system allocator structures a lot of better than just two lock-free lists

### CODE REVIEW

File: core/utils/container_utils.hpp [hunk 1]
```diff
@@ -27,339 +27,146 @@
...
-    }
-
-    // non-precomputed buckets, offset is the sum of previous buckets
-    IRS_ASSERT(
-      !base_t::buffers_
-         .empty());  // otherwise do not know what size buckets to create
-    const auto& bucket =
-      base_t::buffers_.back();  // most of the meta from last computed bucket
-    return base_t::push_buffer(bucket.offset + bucket.size, META.back());
+  const buffer_t& push_buffer() {
```
Comment (Dronplane) [inline core/utils/container_utils.hpp:147]: why it is not in base class? Looks a bit strange that base has "clear" but not "push"

Comment (MBkkt) [reply core/utils/container_utils.hpp:147]: Because it's need META

Comment (Dronplane) [reply core/utils/container_utils.hpp:147]: But then the question - why we need to keep `raw_block_vector_base` ? It is not used anywhere else. Could we just get rid of it and put everything in `raw_block_vector`?

Comment (MBkkt) [reply core/utils/container_utils.hpp:147]: Same as old reason: to instantiate over methods single time

Comment (Dronplane) [reply core/utils/container_utils.hpp:147]: then why not with "push_buffer" part that does not depends on META ?

**Review by Dronplane — APPROVED** [review]
LGTM

**Pull request merged**